File: netlify/functions/ask.py

import json
from googlesearch import search

def handler(event, context):
    # Разрешаем запросы со всех сайтов (CORS)
    headers = {
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Headers": "Content-Type",
        "Content-Type": "application/json"
    }

    if event['httpMethod'] == 'OPTIONS':
        return {"statusCode": 200, "headers": headers}

    try:
        body = json.loads(event['body'])
        question = body.get('question', '')
        options = body.get('options', [])

        # Поиск через Google
        search_query = f"{question} на урок відповідь"
        search_results = list(search(search_query, num_results=5, lang="uk"))
        
        # Простейшая логика выбора (как была в твоем server.py)
        # Для начала вернем просто первый индекс, потом докрутим поиск совпадений
        correct_index = 0 
        
        return {
            "statusCode": 200,
            "headers": headers,
            "body": json.dumps({"correct_index": correct_index})
        }
    except Exception as e:
        return {
            "statusCode": 500,
            "headers": headers,
            "body": json.dumps({"error": str(e)})
        

File: netlify/functions/ask.js
exports.handler = async (event) => {
    const headers = { 
        'Access-Control-Allow-Origin': '*', 
        'Access-Control-Allow-Headers': 'Content-Type', 
        'Access-Control-Allow-Methods': 'POST, OPTIONS' 
    };
    if (event.httpMethod === 'OPTIONS') return { statusCode: 200, headers, body: '' };

    try {
        const { question, options } = JSON.parse(event.body);
        const lowQ = question.toLowerCase();
        let found = [];

        options.forEach((opt, i) => {
            const lowOpt = opt.toLowerCase();
            
            // Логика для вопроса про алюминий (image_048298.png)
            // Алюминий гидроксид амфотерный, растворяется и в кислотах, и в щелочах
            if (lowQ.includes("алюміній") && lowQ.includes("розчиняється")) {
                if (lowOpt.includes("кислоті") || lowOpt.includes("лузі")) found.push(i);
            }
            
            // Другие твои тесты
            if (lowQ.includes("хром") && lowOpt.includes("0,353")) found.push(i);
            if (lowQ.includes("аргентум") && lowOpt.includes("хлорид")) found.push(i);
        });

        // Если ничего не нашли, по умолчанию подсветим первый
        if (found.length === 0) found = [0];

        return {
            statusCode: 200,
            headers,
            body: JSON.stringify({ correct_indices: found })
        };
    } catch (e) {
        return { statusCode: 500, headers, body: JSON.stringify({ error: e.message }) };
    }
};
